pkg/dns/util/util.go

```diff
@@ -42,14 +46,51 @@ const (
 // ExtractIP turns a standard PTR reverse record lookup name
 // into an IP address
 func ExtractIP(reverseName string) (string, bool) {
-	if !strings.HasSuffix(reverseName, ArpaSuffix) {
-		return "", false
+	if strings.HasSuffix(reverseName, ArpaSuffix) {
+		return extractIPv4(strings.TrimSuffix(reverseName, ArpaSuffix))
+	}
+
+	if strings.HasSuffix(reverseName, ArpaSuffixV6) {
+		return extractIPv6(strings.TrimSuffix(reverseName, ArpaSuffixV6))
 	}
-	search := strings.TrimSuffix(reverseName, ArpaSuffix)
 
+	return "", false
+}
+
+// extractIPv4 turns a standard PTR reverse record lookup name
+// into an IP address
+func extractIPv4(reverseName string) (string, bool) {
 	// reverse the segments and then combine them
-	segments := ReverseArray(strings.Split(search, "."))
-	return strings.Join(segments, "."), true
+	segments := ReverseArray(strings.Split(reverseName, "."))
+
+	ip := net.ParseIP(strings.Join(segments, ".")).To4()
+	if ip == nil {
+		return "", false
+	}
+	return ip.String(), true
+}
+
+// extractIPv6 turns a IPv6 PTR reverse record lookup name
+// into an IPv6 address according to RFC3596
+// b.a.9.8.7.6.5.0.4.0.0.0.3.0.0.0.2.0.0.0.1.0.0.0.0.0.0.0.1.2.3.4.ip6.arpa.
+// is reversed to 4321:0:1:2:3:4:567:89ab
```

Comment:
Minor nit for accuracy: the right syntax would have the double colon (::) for the zeros that were shortened?

Reply:
I took the ip value in func comment from RFC

https://datatracker.ietf.org/doc/html/rfc5952#section-4.2.2
```
The symbol "::" MUST NOT be used to shorten just one 16-bit 0 field.
For example, the representation 2001:db8:0:1:1:1:1:1 is correct, but
2001:db8::1:1:1:1:1 is not correct.
```